Network/Socket/Types.hsc

```diff
@@ -820,7 +820,9 @@ withSocketAddress addr f = do
     allocaBytes sz $ \p -> pokeSocketAddress p addr >> f (castPtr p) sz
 
 withNewSocketAddress :: SocketAddress sa => (Ptr sa -> Int -> IO a) -> IO a
-withNewSocketAddress f = allocaBytes 128 $ \ptr -> f ptr 128
+withNewSocketAddress f = allocaBytes 128 $ \ptr -> do
+    zeroMemory ptr 128
```

Comment:
The constant 128 should probably be computed from `max(sizeof(sockaddr_storage), sizeof(sockaddr_un))` or similar (some systems don't have a `sockaddr_storage`, and `sockaddr_in` and `sockaddr_in6` might need to be used instead). And it is here that we might want to add and zero one more byte.

Reply:
Currently, memory is cheap. So, let's just use 128 which is the size of `struct sockaddr_strage`. +1 is not necessary for 128.

Reply:
Done.